File: examples/python/02_search.py
"""Search the US Amazon marketplace through Glade API."""

from glade_client import get


payload = get(
    "/api/amazon/search",
    {
        "searchTerm": "wireless earbuds",
        "domain": "US",
        "sort": "FEATURED",
        "page": 1,
        "limit": 5,
    },
)
search = payload["data"]["amazonProductSearchResults"]
products = (search.get("productResults") or {}).get("results") or []

for index, product in enumerate(products, start=1):
    price = product.get("price") or {}
    print(
        f"{index}. {product.get('title') or 'Untitled'} "
        f"[{product.get('asin') or 'no ASIN'}] — {price.get('display') or 'n/a'}"
    )

File: examples/node/02-search.mjs
import { get } from "./glade-client.mjs";

const payload = await get("/api/amazon/search", {
  searchTerm: "wireless earbuds",
  domain: "US",
  sort: "FEATURED",
  page: 1,
  limit: 5,
});
const products = payload.data.amazonProductSearchResults?.productResults?.results ?? [];

for (const [index, product] of products.entries()) {
  console.log(
    `${index + 1}. ${product.title ?? "Untitled"} [${product.asin ?? "no ASIN"}] — ${product.price?.display ?? "n/a"}`,
  );
}
